crates/analyzer/src/handlers/check_expression.rs

```diff
@@ -36,17 +43,25 @@ impl<'a> VerylGrammarTrait for CheckExpression<'a> {
                 if let Ok(rr) = symbol_table::resolve(expid) {
                     let identifier = rr.found.token.to_string();
                     let token: TokenRange = x.expression_identifier.as_ref().into();
+                    let error = AnalyzerError::invalid_factor(
+                        &identifier,
+                        &rr.found.kind.to_kind_name(),
+                        self.text,
+                        &token,
+                    );
                     match rr.found.kind {
-                        SymbolKind::Function(_)
-                        | SymbolKind::ModportFunctionMember(_)
-                        | SymbolKind::SystemFunction => {
+                        SymbolKind::Function(_) | SymbolKind::ModportFunctionMember(_) => {
                             if x.factor_opt.is_none() {
-                                self.errors.push(AnalyzerError::invalid_factor(
-                                    &identifier,
-                                    &rr.found.kind.to_kind_name(),
-                                    self.text,
-                                    &token,
-                                ));
+                                self.errors.push(error);
+                            } else {
+                                self.call_stack_kind.push(FunctionKind::NonSystem);
```

Comment:
I think `self.call_stack_kind.pop` corresponding to this push may be required at `HandlerPoint::After`.

Reply:
You're correct.